app/src/main/java/ai/elimu/vitabu/ui/storybook/WordAdapter.java

```diff
@@ -0,0 +1,157 @@
+package ai.elimu.vitabu.ui.storybook;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.android.flexbox.FlexboxLayoutManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import ai.elimu.model.v2.gson.content.WordGson;
+import ai.elimu.vitabu.R;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+
+class WordAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    public static final int WORD_TYPE = 0;
+    public static final int NEW_PARAGRAPH_TYPE = 1;
+
+    private final int readingLevelPosition;
+
+    private final OnItemClickListener listener;
+    private final List<String> wordsInOriginalText = new ArrayList<>();
+    private final List<WordGson> wordAudios = new ArrayList<>();
```

Comment:
```suggestion
    private final List<WordGson> wordGsons = new ArrayList<>();
```

I would rename from `WordAudio` to `Word` because a Word does not necessarily have a corresponding Audio recording. In this case, there would be no Audio recording to play when pressed, and we would fall-back to Text-to-Speech.

This gets a bit confusing, so maybe we should refactor the way this works. I'll give it some thought.